src/components/Settings.vue

```diff
@@ -1,41 +1,62 @@
 <script>
 import Toggle from "@components/Toggle.vue"
-import { INFINITE_SCROLL_KEY, SETTINGS_COMMENTS_KEY, writeStorageData, readStorageKey } from "@js/chrome-storage"
+import { EXTENSION_ENABLED_KEY, INFINITE_SCROLL_KEY, SETTINGS_COMMENTS_KEY, writeStorageData, readStorageKey } from "@js/chrome-storage"
 
 export default {
   components: {
     Toggle
   },
   data() {
     return {
+      showExtensionToggle: false,
       showCommentsToggle: false,
+      showInfiniteScrollToggle: false,
+      extensionEnabled: false,
       commentsSectionEnabled: false,
       infiniteScrollEnabled: false,
-      showInfiniteScrollToggle: false
     }
   },
   mounted() {
-    readStorageKey(SETTINGS_COMMENTS_KEY, (value) => {
-      if(typeof(value) === "undefined") {
-        this.commentsSectionEnabled = false
-      } else {
-        this.commentsSectionEnabled = value
+    readStorageKey(EXTENSION_ENABLED_KEY, (value) => {
+      this.extensionEnabled = value !== undefined ? value : false
+      this.showExtensionToggle = true
+      if (this.extensionEnabled) {
+        this.enableOtherSettings()
+      }else {
+        this.disbleOtherSettings()
       }
-
-      this.showCommentsToggle = true
-    })
-
-    readStorageKey(INFINITE_SCROLL_KEY, (value) => {
-      if(typeof(value) === "undefined") {
-        this.infiniteScrollEnabled = false
-      } else {
-        this.infiniteScrollEnabled = value
-      }
-
-      this.showInfiniteScrollToggle = true
     })
   },
   methods: {
+    disbleOtherSettings(){
+      this.commentsSectionEnabled = false
+      this.infiniteScrollEnabled = false
+      this.showCommentsToggle = false
+      this.showInfiniteScrollToggle = false
+
+      
+    },
+    enableOtherSettings() {
+      readStorageKey(SETTINGS_COMMENTS_KEY, (value) => {
+        this.commentsSectionEnabled = value !== undefined ? value : false
+        this.showCommentsToggle = true
+      })
+
+      readStorageKey(INFINITE_SCROLL_KEY, (value) => {
+        this.infiniteScrollEnabled = value !== undefined ? value : false
+        this.showInfiniteScrollToggle = true
+      })
+    },
+    handleExtensionToggle(val) {
+      writeStorageData(EXTENSION_ENABLED_KEY, val, () => {
+        this.extensionEnabled = val
+        if (val) {
```

Comment:
🔍 I think we have to do a full page refresh here, since we've deleted parts of the DOM. Do you think there's a better option?

💡 I think with page refresh we can skip `disbleOtherSettings`.

Reply:
There is a better way. Some extensions make changes dynamically. I'll try implementing that when I get some time.